crates/uv-toolchain/src/discovery.rs

```diff
@@ -934,6 +935,72 @@ impl ToolchainRequest {
         // e.g. foo.exe
         Self::ExecutableName(value.to_string())
     }
+
+    /// Check if a given interpreter satisfies the interpreter request.
+    pub fn satisfied(&self, interpreter: &Interpreter) -> bool {
+        match self {
+            ToolchainRequest::Any => true,
+            ToolchainRequest::Version(version_request) => {
+                version_request.matches_interpreter(interpreter)
+            }
+            ToolchainRequest::Directory(directory) => {
+                // `sys.prefix` points to the venv root.
+                is_same_file(directory, interpreter.sys_prefix()).unwrap_or(false)
+            }
+            ToolchainRequest::File(file) => {
+                // The interpreter satisfies the request both if it is the venv ...
+                if is_same_file(interpreter.sys_executable(), file).unwrap_or(false) {
```

Comment:
Nit: can we reverse the order of these arguments to match the `ToolchainRequest::Directory` case?